app/models/user.rb

```diff
@@ -192,6 +192,30 @@ def most_accurate_school_country
     school.present? ? school.country : 'United States'
   end
 
+  def school_location
+    return UserHelper.convert_to_user_location(school.location) if school.present? && school.has_attribute?("location")
```

Comment:
`has_attribute?` will return false if, for example, the column doesn't exist or if the school was loaded with `School.select(...)` and the select did not include location, leading to inconsistent behavior. I suggest using something like `!school.location.blank?` which would throw an error instead in those cases.

Reply:
@Dantemss I was using school.location.blank?, but a test was failing because it was a school with only an id.

Reply:
Fixed